airbyte-workers/src/main/java/io/airbyte/workers/temporal/TemporalClient.java

```diff
@@ -446,6 +451,79 @@ public ManualOperationResult synchronousResetConnection(final UUID connectionId,
         Optional.of(resetJobId), Optional.empty());
   }
 
+  public void restartWorkflowByStatus(final WorkflowExecutionStatus executionStatus) {
+    final Set<String> workflowExecutionInfos = fetchWorkflowByStatus(executionStatus);
+
+    final Set<String> nonRunningWorkflow = filterOutRunningWorkspaceId(workflowExecutionInfos);
```

Comment:
How large can these sets be?  Are we worried about running out of memory?  Should we be iterating in batches? 

Maybe the method signature could be `fetchWorkflowByStatus(executionStatus, limit=100)` and we keep calling it until we get 0 results.  In the method below, you are already paging though results with `nextPagetoken` - perhaps we don't need to do that.

Reply:
The limiting number here is number of connection ID. I don't expect it to be a concern in the short term.

In order to limit the memory consumption, I will change this to use UUID instead of String.

Reply:
I'm still concerned about this method returning quite a lot of data once we have 1000s of workspaces and connections, but I agree for now this is probably OK.